src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -364,32 +364,54 @@ private static string ConvertIcuTimeFormatString(ReadOnlySpan<char> icuFormatStr
                 char current = icuFormatString[i];
                 switch (current)
                 {
+                    case '\\':
```

Comment:
Could you please add a little comment telling ICU formats don't have escaping so we need to escape `\` if we encountered any?